### PR TITLE
fix: resize challenge and organization cards

### PR DESCRIPTION
**Pull request by vpchung**
Related to #695 and should be reviewed after merging #709 (this branch is based on `fix-layout-size` rather than `HEAD`).

Remaining TODOs:
* difficulty icon is missing now for some reason
* test out varying sized card titles/subtitles/descriptions - do they wrap appropriately?
* fix layout of card group, e.g. this should be >1 card per row
Screen Shot 2022-09-16 at 12 08 29 PM

### CODE REVIEW

File: libs/challenge-registry/ui/src/lib/challenge-card/challenge-card.component.scss [hunk 1]
```diff
@@ -1,181 +1,70 @@
...
-  top: 456px;
-  left: 38px;
-  width: 420px;
-  height: 41px;
+.difficulty-tag {
   display: flex;
-  position: absolute;
-  border-top: 1px solid #c4c4c4;
-  align-items: flex-start;
+  flex-direction: column;
```
Comment (rrchai) [inline libs/challenge-registry/ui/src/lib/challenge-card/challenge-card.component.scss:53]: @vpchung  here is the reason why the `timeline` icon is missing.

File: libs/challenge-registry/user-profile/src/lib/user-profile-starred/user-profile-starred.component.html [hunk 2]
```diff
@@ -1,32 +1,40 @@
...
-          [organization]="organization"
-        >
-        </challenge-registry-organization-card>
-      </div>
-    </ng-container>
-    <div class="main-section-stars-paginator"></div>
-  </section>
+    <div class="card-group">
+      <ng-container [ngSwitch]="activeTab">
+        <div *ngSwitchCase="tabs['challenges']" class="main-section-stars-card-group">
```
Comment (rrchai) [inline libs/challenge-registry/user-profile/src/lib/user-profile-starred/user-profile-starred.component.html:22]: @vpchung  challenge card group is actually this selector, `main-section-stars-card-group`. You are applying the flex to its parent `card-group`. Fixing this should make the challenge cards responsive :)

Comment (vpchung) [reply libs/challenge-registry/user-profile/src/lib/user-profile-starred/user-profile-starred.component.html:22]: Ah, good eye! Thanks, @rrchai :-)

**Comment by vpchung** [comment]
I tested a longer title and description in the challenge card:

Screen Shot 2022-09-16 at 9 56 49 PM

IIRC, there was a previous conversation ages ago that ideally, all cards should be the same height regardless of content..? If so, I will set a max-height.

File: libs/challenge-registry/ui/src/lib/challenge-card/challenge-card.component.scss (hunk 1 above)
Comment (vpchung) [reply libs/challenge-registry/ui/src/lib/challenge-card/challenge-card.component.scss:53]: Darn my copy-paste error.... thanks!

File: libs/challenge-registry/ui/src/lib/challenge-card/challenge-card.component.scss [hunk 3]
```diff
@@ -1,181 +1,74 @@
...
-  left: 323px;
-  width: 135px;
-  height: 41px;
+.difficulty-tag {
   display: flex;
-  position: absolute;
+  flex: 1;
+  height: 24px;
+}
+.status-tag {
```
Comment (rrchai) [inline libs/challenge-registry/ui/src/lib/challenge-card/challenge-card.component.scss:62]: suggest to add some height or padding in the status tag, i.e. :
current: Screen Shot 2022-09-23 at 12 15 40 PM
adding `padding: 6px 0;`: Screen Shot 2022-09-23 at 12 15 28 PM

File: libs/challenge-registry/user-profile/src/lib/user-profile-overview/user-profile-overview.component.html [hunk 4]
```diff
@@ -1,16 +1,14 @@
...
-    </div>
-    <div class="main-section-bio-container2">
-      <h3 class="main-section-bio-text7">ORGANIZATIONS</h3>
-      <challenge-registry-organization-card
+<main *ngIf="user">
+  <div id="bio" class="row col">
+    <h3>Biography</h3>
+    <p>{{ user.bio }}</p>
+  </div>
+  <div id="orgs" class="row">
```
Comment (rrchai) [inline libs/challenge-registry/user-profile/src/lib/user-profile-overview/user-profile-overview.component.html:6]: I think `col` is needed here instead?
Screen Shot 2022-09-23 at 12 38 11 PM

**Review by rrchai — CHANGES_REQUESTED** [review]

**Comment by rrchai** [comment]
> IIRC, there was a previous conversation ages ago that ideally, all cards should be the same height regardless of content..? If so, I will set a max-height.
@vpchung If I remember correctly, we decided to make the height of stretchable for the list layout? Not sure about the grid layout, though.
We might have the discussion about limiting text/width length in the description in `challenge-card` and use ellipsis to leave out extra words.

File: libs/challenge-registry/ui/src/lib/challenge-card/challenge-card.component.scss [hunk 5]
```diff
@@ -1,181 +1,74 @@
...
+  margin: 0 2px;
 }
-.challenge-card-card-title {
-  top: 194px;
-  left: 75px;
-  width: 347px;
-  height: 86px;
+
+// BODY
+.card-body {
```
Comment (rrchai) [inline libs/challenge-registry/ui/src/lib/challenge-card/challenge-card.component.scss:38]: This needs to be fixed:
Screen Shot 2022-09-23 at 12 26 42 PM

Comment (vpchung) [reply libs/challenge-registry/ui/src/lib/challenge-card/challenge-card.component.scss:38]: 🤣

File: libs/challenge-registry/user-profile/src/lib/user-profile-overview/user-profile-overview.component.html [hunk 6]
```diff
@@ -1,16 +1,14 @@
...
-  <section id="main" class="main-section-bio-main-section">
-    <div class="main-section-bio-container1">
-      <span class="main-section-bio-text">Biography</span>
-      <span class="main-section-bio-text1">{{ user.bio }}</span>
-    </div>
-    <div class="main-section-bio-container2">
-      <h3 class="main-section-bio-text7">ORGANIZATIONS</h3>
-      <challenge-registry-organization-card
+<main *ngIf="user">
+  <div id="bio" class="row col">
```
[the remark this answers is not in the document]
Comment (vpchung) [reply libs/challenge-registry/user-profile/src/lib/user-profile-overview/user-profile-overview.component.html:2]: .... it was me being lazy and not wanting to declare another div 😅  It's essntially shorthand for:

```
<div id="bio" class="row">
    <div class="col">
      <h3>Biography</h3>
      <p>{{ user.bio }}</p>
    </div>
  </div>
```

**Review by rrchai — APPROVED** [review]
LGTM :rocket:

**Pull request merged**